geomesa-convert/geomesa-convert-xml/src/test/scala/org/locationtech/geomesa/convert/xml/XmlCompositeConverterTest.scala

```diff
@@ -0,0 +1,100 @@
+/***********************************************************************
+ * Copyright (c) 2013-2019 Commonwealth Computer Research, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Apache License, Version 2.0
+ * which accompanies this distribution and is available at
+ * http://www.opensource.org/licenses/apache2.0.php.
+ ***********************************************************************/
+
+package org.locationtech.geomesa.convert.xml
+
+import java.io.ByteArrayInputStream
+import java.nio.charset.StandardCharsets
+
+import com.typesafe.config.ConfigFactory
+import org.junit.runner.RunWith
+import org.locationtech.geomesa.convert2.SimpleFeatureConverter
+import org.locationtech.geomesa.utils.geotools.SimpleFeatureTypes
+import org.locationtech.geomesa.utils.io.WithClose
+import org.locationtech.geomesa.utils.text.WKTUtils
+import org.specs2.mutable.Specification
+import org.specs2.runner.JUnitRunner
+
+@RunWith(classOf[JUnitRunner])
+class XmlCompositeConverterTest extends Specification {
+
+  sequential
+
+  val sft = SimpleFeatureTypes.createType(
+    ConfigFactory.parseString(
+      """{ type-name = "xmlFeatureType"
+        |  attributes = [
+        |    { name = "number", type = "Integer" }
+        |    { name = "color",  type = "String"  }
+        |    { name = "weight", type = "Double"  }
+        |    { name = "geom",   type = "Point"   }
+        |  ]
+        |}
+      """.stripMargin)
+  )
+
+  "XmlCompositeConverter" should {
+
+    "select a parser based on a predicate" >> {
+      val xmlStr =
+        """<doc><type>a</type><id>1</id><number>123</number><color>red</color><physical><weight>127.5</weight><height>5'11"</height></physical><lat>0</lat><lon>0</lon></doc>
```

Comment:
Does XML composite converters work by applying different converters to each file?  Or can one process sub-documents differently?

Reply:
you can only apply a single converter to each xml doc, so you couldn't use a feature-path and handle each sub-document separately, if that's what you're asking.

Reply:
if that's a use case, we can open a ticket to support it.